manager/traverse.go

```diff
@@ -9,12 +9,73 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/rancher/catalog-service/helm"
 	"github.com/rancher/catalog-service/model"
 	"github.com/rancher/catalog-service/parse"
 	"gopkg.in/yaml.v2"
 )
 
-func traverseFiles(repoPath string) ([]model.Template, []error, error) {
+func traverseFiles(repoPath, kind string) ([]model.Template, []error, error) {
+	if kind == "" || kind == "rancher" {
+		return traverseGitFiles(repoPath)
+	}
+	if kind == "k8s" {
+		index, err := helm.LoadIndex(repoPath)
```

Comment:
Can you move the stuff under `if kind == "k8s" {` into a separate function like it is for `rancher`?